spdx/parsers/jsonyamlxml.py

```diff
@@ -1495,6 +1495,34 @@ def parse_pkg_chksum(self, pkg_chksum):
             self.value_error("PKG_CHECKSUM", pkg_chksum)
 
 
+def flatten_document(document):
+    """
+    Flatten document to match current data model. File objects are nested within packages according to hasFiles-tag.
+    """
+    files_by_id = {}
+    if "files" in document:
+        for f in document.get("files"):
+            # XXX must downstream rely on "sha1" property?
+            for checksum in f["checksums"]:
+                if checksum["algorithm"] == "SHA1" or "sha1" in checksum["algorithm"]:
+                    f["sha1"] = checksum["checksumValue"]
+                    break
+            files_by_id[f["SPDXID"]] = f
+    if "packages" in document:
+        packages = document.get("packages")
+        for package in packages:
+            if "hasFiles" in package:
+                package["files"] = [{
+                    "File": files_by_id[spdxid.split("#")[-1]]} for spdxid in package["hasFiles"]
```

Comment:
At this point the ID is only used for the mapping. Since the jsonyamlxml-writer discard the part before `#` the keys in the dict `files_by_id` only contain the SPDX Identifier. To match these values we need to split the SPDXID in hasFiles since they can still contain a namespace or URI.